### PR TITLE
Push animal changes.

### PR DESCRIPTION
**Pull request by tclaar**

### CODE REVIEW

**Review by b1gd3vd0g — APPROVED** [review]
This is great! I don't really have too many notes.

File: js/classes/entities/foreground/health_bar.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+     * Constructor for a HealthBar. The target must be an entity with a field named "health" that
+     * is a number.
+     * 
+     * @param {Entity} target the entity to assign to the HealthBar (must have a health field)
+     * @param {number} targetMaxHealth the maximum health of target
+     * @param {number} targetWidth the width of target
+     * @param {number} [barWidth] (OPTIONAL) the width of the HealthBar
+     * @throws {Error} Will throw an error if target has no numeric health field.
+     */
+    constructor(target, targetMaxHealth, targetWidth, barWidth = HealthBar.DEFAULT_SIZE.x) {
```
Comment (calebkrauter) [inline js/classes/entities/foreground/health_bar.js:17]: I like how generic this is, we can reuse this a lot!

Comment (calebkrauter) [inline js/classes/entities/foreground/health_bar.js:17]: Didn't know you could initialize a parameter like this! Cool.

**Review by calebkrauter — COMMENTED** [review]
Looks great!

**Review by calebkrauter — APPROVED** [review]
Good work!

**Pull request merged**